### ISSUE TITLE
Update entity does not show update if use latest image

### ISSUE THREAD

**Issue by royto** [fmartinou/whats-up-docker#191]
I have configured WUD with HA.

Most of my images use the tag latest. Since the 6.0.0 version the new update entity in HA don't show update ...

It was working well with previous version

example with nextcloud 
![image](https://user-images.githubusercontent.com/6990995/207368569-a416a923-8270-46df-bd00-20451a294dda.png)

In HA
![image](https://user-images.githubusercontent.com/6990995/207368720-93759e9e-fc76-4fa7-b4ae-14c85adba77c.png)

I had one container with a semver version and it is working

![image](https://user-images.githubusercontent.com/6990995/207368989-87864a65-92ad-4b16-8b4d-9dc97dd95186.png)

![image](https://user-images.githubusercontent.com/6990995/207369051-dd931ca7-aa9c-4ea8-89a6-fca78ce5b880.png)

**Comment by adorobis** [comment]
Same here. Additionally the binary_sensors sometimes disappear so I don't get notified when an update is available.

**Comment by fmartinou** [comment]
Hi,

That's a regression compared to the previous binary_sensors, yes.

The problem I see it's that HA computes the state on its end (update available / no update available) by comparing the 2 values (installed_version / latest version).

So to make it work with `latest` images, I need to provide `sha256` values as current ans latest so that HA can check if both values are equal/different.

You won't see `latest` anymore on HA; would that be a problem?

**Comment by adorobis** [comment]
No, I think it makes more sense actually. You could put the tag into a different field that is available for MQTT Update integration e.g. `release_summary` or `title`. I think they are not in use at the moment. Also filling in the `release_url` would be great (although I'm not sure if it is difficult to figure it out from e.g. docker hub).

**Comment by adorobis** [comment]
On another note, is it not possible to get a version number of the running container? That would be ideal :)

**Comment by pavolholes** [comment]
I've just briefly looked into [Home Assistant documentation for MQTT Update](https://www.home-assistant.io/integrations/update.mqtt/#examples) and I've documented my thinking below and also with the suggested fix at the end.

AdGuard example: entity `update.wud_container_local_adguard`
Topic `homeassistant/update/wud_container_local_adguard/config`
```
{
  ...
  "value_template": "{{ value_json.image_tag_value }}",
  "latest_version_template": "{{ value_json.result_tag or value_json.result_digest }}",
  ...
}

```
and now from the state topic `wud/container/local/adguard` I will pick only what will be used in above templates:
```
{
  ...
  "image_tag_value": "latest",
  "result_tag": "latest",
  "result_digest":   "sha256:3b03d39541bc024b073c88595e09426071cb2877656e9e3b3f54b86c9a4a7e2c"
  ...
}
```
When I combine them I get the result:
```
  "value_template": "latest"
  "result_tag": "latest"
```
which is equal so the entity state is set to `off`. Even though the `digest` update is available in the value:
```
"update_kind_kind": "digest"
```

When I look at the example from the MQTT Update in the docs for the Shelly 1 I see they are using more complicated logic:
```
latest_version_template: "{% if value_json['update'].new_version %}{{ value_json['update'].new_version }}{% else %}{{ value_json['update'].old_version }}{% endif %}"
```
which if I understant correctly only backfill the template with old_version if new_version doesn't exist. But that gave me an idea:

We have `update_kind_kind` value which I was able to track is set to one of:
- digest
- tag
- unknown

So I would suggest a new logic for the `latest_version_template` as below:
```
{% if value_json.result_tag == "digest" %}{{ value_json.result_digest[:15] }}
{% elif value_json.result_tag == "tag" %}{{ value_json.result_tag }}
{% else %}{{ value_json.result_tag }}
{% endif %}
```
which can actually be simplified to:
```
{% if value_json.result_tag == "digest" %}{{ value_json.result_digest[:15] }}
{% else %}{{ value_json.result_tag }}
{% endif %}
```
Note: `[:15]` should shorten the digest to 15 characters to keep it consistent with what we can see in the WUD GUI.

I've also looked into Zigbee2MQTT what is their logic and it's like this:
```
latest_version_template: >-
  {% if value_json['update']['state'] == "available" %}{{ 'newer' }}
  {% else %}{{value_json['update']['installed_version'] }}
  {% endif %}
```
So when there is some new update they simply set the template to fixed value `newer` 😁.

I think my suggested solution is better then in Zigbee2MQTT.

Please let me know what do you think about the suggested fix.

Thank you!

**Comment by adorobis** [comment]
@pavolholes excellent analysis! I've tested it and have a small correction: the tested condition should be the `value_json.update_kind_kind == "digest"` so the final would look like:
```
latest_version_template: >-
  {% if value_json.update_kind_kind == "digest" %}
  {{ value_json.result_digest[:15] }}{% else %}
  {{ value_json.result_tag }}
  {% endif %}
```
To make it work it is just one line correction in the Mqtt.js:
`const hassLatestVersionTemplate = '{% if value_json.update_kind_kind == "digest" %}{{ value_json.result_digest[:15] }}{% else %}{{ value_json.result_tag }}{% endif %}';`

It works great not only for `latest` image tags but also for others e.g. `dev`

**Comment by pavolholes** [comment]
Hi @adorobis, ah you're right! I've made a copy&paste mistake, it was already 1:30AM and I was tired when I was posting the comment. Thanks for checking and testing, I'm glad it's working. 
Now @fmartinou can look at it and hopefully implement :)
Have a great weekend everyone ;)

**Comment by fmartinou** [comment]
Hi guys,

Here is the dev version including the fix on the `latest_version_template`
```
fmartinou/whats-up-docker:fix__191_fix_ha_state_for_nonsemver_tags
```

May you give it a try?
If it's ok, then, I'll release it.

Thanks!

**Comment by adorobis** [comment]
works very well, many thanks!

**Comment by fmartinou** [comment]
Thanks for the feedback! 👍 

It's just released as the latest `6.0.1`!

**Comment by adorobis** [comment]
Cool! Thank you for quick fix! 👌
